be/src/http/action/transaction_stream_load.cpp

```diff
@@ -132,6 +134,38 @@ void TransactionManagerAction::handle(HttpRequest* req) {
     _send_reply(req, resp);
 }
 
+// Handle the resource acquired by the http request
+class ResourceHandler {
+public:
+    // ctx has been referenced and locked outside
+    ResourceHandler(StreamLoadContext* ctx) : _ctx(ctx) {
+        DCHECK(_ctx != nullptr);
+        DCHECK(!_ctx->lock.try_lock());
+    }
+
+    ~ResourceHandler() { release(); }
+
+    StreamLoadContext* ctx() { return _ctx; }
+
+    void release() {
+        if (_released) {
+            return;
+        }
+        _released = true;
+        _ctx->lock.unlock();
+        if (config::enable_stream_load_verbose_log) {
+            LOG(INFO) << "release resource, " << _ctx->brief();
+        }
+        if (_ctx->unref()) {
+            delete _ctx;
```

Comment:
reset the _ctx to nullptr, to prevent any unexpected access to this ctx after release()

Reply:
fixed